src/eventgenerator/generator/applicationManager.go

```diff
@@ -0,0 +1,78 @@
+package generator
+
+import (
+	"code.cloudfoundry.org/clock"
+	"code.cloudfoundry.org/lager"
+	"eventgenerator/model"
+	"sync"
+	"time"
+)
+
+type ConsumeAppMonitorMap func(map[string][]*model.AppEvaluationEntity, chan []*model.AppEvaluationEntity)
+type ApplicationManager struct {
+	evaluateInterval             time.Duration
+	logger                       lager.Logger
+	cclock                       clock.Clock
+	lock                         *sync.Mutex
+	ticker                       clock.Ticker
+	stopping                     bool
+	doneChan                     chan bool
+	appEvaluationEntityMapChan   chan map[string][]*model.AppEvaluationEntity
+	appEvaluationEntityArrayChan chan []*model.AppEvaluationEntity
+	appEvaluationEntityArrayMap  map[string][]*model.AppEvaluationEntity
+}
+
+func NewApplicationManager(evaluateInterval time.Duration, logger lager.Logger, cclock clock.Clock, appEvaluationEntityMapChan chan map[string][]*model.AppEvaluationEntity, appEvaluationEntityArrayChan chan []*model.AppEvaluationEntity) *ApplicationManager {
+	return &ApplicationManager{
+		evaluateInterval:             evaluateInterval,
+		logger:                       logger.Session("ApplicationManager"),
+		cclock:                       cclock,
+		lock:                         &sync.Mutex{},
+		stopping:                     false,
+		doneChan:                     make(chan bool),
+		appEvaluationEntityMapChan:   appEvaluationEntityMapChan,
+		appEvaluationEntityArrayChan: appEvaluationEntityArrayChan,
+		appEvaluationEntityArrayMap:  make(map[string][]*model.AppEvaluationEntity),
+	}
+}
+func (a *ApplicationManager) Start() {
+	a.ticker = a.cclock.NewTicker(a.evaluateInterval)
+	go a.startWork()
+}
+func (a *ApplicationManager) startWork() {
+	for {
+		a.AddApplicationToEvaluate(a.appEvaluationEntityArrayMap, a.appEvaluationEntityArrayChan)
+		select {
+		case <-a.doneChan:
+			return
+		case appEvaluationEntityMapTmp := <-a.appEvaluationEntityMapChan:
+			a.refreshAppEvaluationEntityMap(appEvaluationEntityMapTmp)
+		case <-a.ticker.C():
+		}
+	}
+	a.logger.Info("ApplicationManager started")
+}
+func (a *ApplicationManager) Stop() {
+	a.stopping = true
```

Comment:
Why do we even have this? It is only checked in one place.
It is also a race issue.